lib/py/src/transport/sslcompat.py

```diff
@@ -66,6 +66,14 @@ def legacy_validate_callback(cert, hostname):
 
```

Comment:
In the Ubuntu 14.04 LTS docker image, Python is version 2.7.6.  This means it doesn't have SSLContext.  This code detects that condition and sets a single version that it can use.  I am actually going to change the default to TLSv1.2 for security purposes here.  Any language that cannot properly modify the openssl options to negotiate TLSv1.0 "or higher" should be forced to TLSv1.2.  From what I can tell, csharp, d, and py2 (in the build image) lack the necessary code/knowledge to allow for a SSLv23 handshake and then autonegotiate TLSv1.0 "or higher".